### PR TITLE
Issue Resolved #296

### LINKED ISSUE #296: Add new Bookmark page throws error when a previously existing URL is added.

### PR DESCRIPTION
**Pull request by rishirajsinghjhelumi**

### CODE REVIEW

**Comment by mitechie** [comment]
This looks like a really solid change. Are you up to trying to write a test for it? We should add a test that verifies that you get the message when posting a url that already exists. This would go with the other bmark tests in https://github.com/bookieio/Bookie/blob/develop/bookie/tests/test_webviews/test_webviews.py#L86

**Comment by rishirajsinghjhelumi** [comment]
I will surely write tests for it. I will go through the above files and get back to you in case I don't understand anything.

File: bookie/tests/test_webviews/test_webviews.py [hunk 1]
```diff
@@ -108,6 +108,38 @@ def test_manual_entry_error(self):
             })
         self.assertIn('not valid', res.body)
 
+    def test_existing_url_entry_error(self):
+        """ Verify whether the User has recieved the message that a previous URL already exists """
+        self._login_admin()
+
+        _test_url = u"http://bmark.us/test"
```
Comment (mitechie) [inline bookie/tests/test_webviews/test_webviews.py]: there's no need to underscore these two lines.

File: bookie/tests/test_webviews/test_webviews.py [hunk 2]
```diff
@@ -108,6 +108,38 @@ def test_manual_entry_error(self):
...
+        _existing_url_message = "URL already Exists"
+
+        #Add The Bookmark Once
```
Comment (mitechie) [inline bookie/tests/test_webviews/test_webviews.py]: always have a space after the # in the comment and make it a real sentence. 

# Add the bookmark once.

File: bookie/tests/test_webviews/test_webviews.py [hunk 3]
```diff
@@ -108,6 +108,38 @@ def test_manual_entry_error(self):
...
+                'description': '',
+                'extended': '',
+                'tags': ''
+            })
+        self.assertEqual(
+            res.status,
+            "302 Found",
+            msg='recent status is 302 Found, ' + res.status)
+
+        #Add the Bookmark Again
```
Comment (mitechie) [inline bookie/tests/test_webviews/test_webviews.py]: # Add the bookmark again.

File: bookie/views/bmarks.py [hunk 4]
```diff
@@ -165,6 +165,17 @@ def edit_error(request):
     with ReqAuthorize(request, username=rdict['username']):
         if 'new' in request.url:
             try:
+                try:
+                    bmark = BmarkMgr.get_by_url(post['url'])
+                except:
+                    bmark = None
+                if bmark:
```
Comment (mitechie) [inline bookie/views/bmarks.py:172]: can we tweak the spacing here? It should be four spaces indent and it looks off here. All indentation should be spaces. Maybe there's some tabs in here?

**Comment by mitechie** [comment]
Landed, thanks for the changes!

**Pull request closed without merging**